tvb_library/tvb/simulator/models/k_ion_exchange.py

```diff
@@ -343,3 +345,126 @@ def V_dot_form(I_Na,I_K,I_Cl,I_pump):
+
+    # Constants
+    Cnap = 21.0  # mol.m**-3
+    DCnap = 2.0  # mol.m**-3
+    Ckp = 5.5  # mol.m**-3
+    DCkp = 1.0  # mol.m**-3
+    Cmna = -24.0  # mV
+    DCmna = 12.0  # mV
+    Chn = 0.4  # dimensionless
+    DChn = -8.0  # dimensionless
+    Cnk = -19.0  # mV
+    DCnk = 18.0  # mV #Ok in the paper
+    g_Cl = 7.5  # nS #Ok in the paper   # chloride conductance
+    g_Na = 40.0  # nS   # maximal sodiumconductance
+    g_K = 22.0  # nS  # maximal potassium conductance
+    g_Nal = 0.02  # nS  # sodium leak conductance
+    g_Kl = 0.12  # nS  # potassium leak conductance
+    rho = 250.  # 250.,#pA # maximal Na/K pump current
+    w_i = 2160.0  # umeter**3  # intracellular volume
+    w_o = 720.0  # umeter**3 # extracellular volume
+    Na_i0 = 16.0  # mMol/m**3 # initial concentration of intracellular Na
+    Na_o0 = 138.0  # mMol/m**3 # initial concentration of extracellular Na
+    K_i0 = 130.0  # mMol/m**3 # initial concentration of intracellular K
+    K_o0 = 4.80  # mMol/m**3 # initial concentration of extracellular K
+    Cl_i0 = 5.0  # mMol/m**3 # initial concentration of intracellular Cl
+    Cl_o0 = 112.0  # mMol/m**3 # initial concentration of extracellular Cl
+
+    # helper functions
+
+    def m_inf(V):
+        return 1.0 / (1.0 + numpy.exp((Cmna - V) / DCmna))
+
+    def n_inf(V):
+        return 1.0 / (1.0 + numpy.exp((Cnk - V) / DCnk))
+
+    def h(n):
+        return 1.1 - 1.0 / (1.0 + numpy.exp(-8.0 * (n - 0.4)))
+
+    def I_K_form(V, n, K_o, K_i):
+        return (g_Kl + g_K * n) * (V - 26.64 * numpy.log(K_o / K_i))
+
+    def I_Na_form(V, Na_o, Na_i, n):
+        return (g_Nal + g_Na * m_inf(V) * h(n)) * (V - 26.64 * numpy.log(Na_o / Na_i))
+
+    def I_Cl_form(V):
+        return g_Cl * (V + 26.64 * numpy.log(Cl_o0 / Cl_i0))
+
+    def I_pump_form(Na_i, K_o):
+        return rho * (
+                1.0 / (1.0 + numpy.exp((Cnap - Na_i) / DCnap)) * (1.0 / (1.0 + numpy.exp((Ckp - K_o) / DCkp))))
+
+    def V_dot_form(I_Na, I_K, I_Cl, I_pump):
+        return (-1.0 / Cm) * (I_Na + I_K + I_Cl + I_pump)
+
+    beta = w_i / w_o
+    DNa_i = -DKi
+    DNa_o = -beta * DNa_i
+    DK_o = -beta * DKi
+    K_i = K_i0 + DKi
+    Na_i = Na_i0 + DNa_i
+    Na_o = Na_o0 + DNa_o
+    K_o = K_o0 + DK_o + Kg
+
+    ninf = n_inf(V)
+    I_K = I_K_form(V, n, K_o, K_i)
+    I_Na = I_Na_form(V, Na_o, Na_i, n)
+    I_Cl = I_Cl_form(V)
+    I_pump = I_pump_form(Na_i, K_o)
+
+    r = R_minus[0] * x / numpy.pi
+    Vdot = (-1.0 / Cm[0]) * (I_Na + I_K + I_Cl + I_pump)
+
+    if_xdot = Delta[0] + 2 * R_minus[0] * (V - c_minus[0]) * x - J[0] * r * x
+    else_xdot = Delta[0] + 2 * R_plus[0] * (V - c_plus[0]) * x - J[0] * r * x
+
+    if_Vdot = Vdot - R_minus[0] * x ** 2 + eta[0] + (R_minus[0] / numpy.pi) * Coupling_Term * (E[0] - V)
+    else_Vdot = Vdot - R_plus[0] * x ** 2 + eta[0] + (R_minus[0] / numpy.pi) * Coupling_Term * (E[0] - V)
+
+    dx[0] = numpy.where(V <= (Vstar * numpy.ones_like(V)), if_xdot, else_xdot)[0]
```

Comment:
why is there a `[0]` at the end of this expression?

Reply:
the np.where was removed and so the [0]